toolchain/sem_ir/function.h

```diff
@@ -38,17 +38,28 @@ struct FunctionFields {
   // the return slot.
   //
   // The parameters appear in declaration order: `self` (if present), then the
-  // explicit runtime parameters, then the return slot (which is "declared" by
-  // the function's return type declaration). This is not populated on imported
-  // functions, because it is relevant only for a function definition.
+  // explicit runtime parameters, then the return parameters (which are
+  // "declared" by the function's return type declaration). This is not
+  // populated on imported functions, because it is relevant only for a function
+  // definition.
   InstBlockId call_params_id;
 
-  // A reference to the instruction in the entity's pattern block that depends
-  // on all other pattern insts pertaining to the return slot pattern. This may
-  // or may not be used by the function, depending on whether the return type
-  // needs a return slot, but is always present if the function has a declared
-  // return type.
-  InstId return_slot_pattern_id;
+  // The type inst representing the function's explicitly declared return type,
+  // if any.
```

Comment:
Maybe this should be:

```suggestion
  // The type inst representing the function's return type, as computed from
  // an explicit return pattern (if any).
```

Or maybe s/pattern/form/ ?

Just trying to get at the idea that the _type_ may not be what is explicit.

Reply:
Right now, the type _is_ what's explicit, because we don't yet support return forms, or even `ref` returns. Of course, this comment will be wrong later when we add support for those things, but at that point the field's name and type will probably also be wrong, so it seems simplest to update all three together.

Also, "as computed from" suggests to me that this inst is the result of constant-evaluating the return type expression, but this has to be the inst that represents the return type expression the user actually wrote, because it's what we use when we need a `LocId` pointing to that expression.

Reply:
Hmm, I guess it seemed like you were trying to move away from the type-only model of returns in this PR, and so expected this to be moving away already.

But if that's a future thing, makes sense.

Reply:
This PR is certainly starting to enter a transitional state between the type-only model and the forms model, so it's probably a bit awkward and inconsistent, but I'm trying to minimize that where I can.